### PR TITLE
[SPARK-10386] [MLlib] PrefixSpanModel supports save/load

### PR DESCRIPTION
**Pull request by yanboliang**
`PrefixSpanModel` supports `save/load`. It's similar with #9267.

cc @jkbradley

### CODE REVIEW

**Comment by jkbradley** [comment]
I'd like to get this into 2.0.  Could you please fix the merge conflicts?  Thanks!

**Comment by yanboliang** [comment]
@jkbradley I have resolved conflicts, this is ready for review now. Thanks!

**Comment by jkbradley** [comment]
Taking a look now

File: mllib/src/test/scala/org/apache/spark/mllib/fpm/PrefixSpanSuite.scala [hunk 1]
```diff
@@ -357,6 +358,39 @@ class PrefixSpanSuite extends SparkFunSuite with MLlibTestSparkContext {
...
+      .setMinSupport(0.5)
+      .setMaxPatternLength(5)
+    val model = prefixSpan.run(rdd)
+
+    val tempDir = Utils.createTempDir()
+    val path = tempDir.toURI.toString
+    try {
+      model.save(sc, path)
+      val newModel = PrefixSpanModel.load(sc, path)
+      val actual = newModel.freqSequences.collect()
```
Comment (jkbradley) [inline mllib/src/test/scala/org/apache/spark/mllib/fpm/PrefixSpanSuite.scala]: I'd compare with the original model, not an expected result, here.

**Comment by jkbradley** [comment]
This looks good; I just had 1 comment

**Comment by jkbradley** [comment]
LGTM
Merging with master
Thanks @yanboliang !

File: mllib/src/main/scala/org/apache/spark/mllib/fpm/PrefixSpan.scala [hunk 2]
```diff
@@ -566,4 +576,88 @@ object PrefixSpan extends Logging {
...
+   *  - Parquet formatted data to path/data/
+   *
+   * The model may be loaded using [[PrefixSpanModel.load]].
+   *
+   * @param sc  Spark context used to save model data.
+   * @param path  Path specifying the directory in which to save this model.
+   *              If the directory already exists, this method throws an exception.
+   */
+  @Since("2.0.0")
+  override def save(sc: SparkContext, path: String): Unit = {
```
Comment (tedyu) [inline mllib/src/main/scala/org/apache/spark/mllib/fpm/PrefixSpan.scala]: Is sc used ?

Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/mllib/fpm/PrefixSpan.scala]: Not here, but it's needed for other algorithms (with local representations).

**Pull request closed without merging**